### DISCUSSION TITLE
init_app: Not initializing migrations (flask + sqlite + alchemy + migrations + socketio)

### DISCUSSION THREAD

**Discussion by ManiMatter** [miguelgrinberg/Flask-Migrate#531] [Q&A] [answered]
hi All,

I am trying to create a socket.io backend with an sqlite DB, supported by flask_sqlalchemy + flask_migrate.

Where I am stuck is this:
- When I run "flask db init" manually via the console, the db is set up as it should and the migration scripts are created
- When I run "flask run", neither db nor migration scripts are created (eventhough db.init_app and migrate.init_app are in the script), but the rest of the code works

Any pointers would be highly appreciated! Thank you for your time!

Here is my app/__init__.py
```
from flask import Flask
from config import Config
from flask_sqlalchemy import SQLAlchemy
from flask_migrate import Migrate
from flask_socketio import SocketIO

socketio = SocketIO()
db = SQLAlchemy()
migrate = Migrate()

def create_app(debug=False):
    """Create an application."""
    app = Flask(__name__)
    app.config.from_object(Config)
    db.init_app(app)
    migrate.init_app(app, db)

    # API Endpoints + Script
    from .main import main as main_blueprint
    app.register_blueprint(main_blueprint)

    # DB Setup
    from app import models

    socketio.init_app(app)
    return app
```

Here is my app/config.py:

```
import os
basedir = os.path.abspath(os.path.dirname(__file__))

class Config(object):
    SECRET_KEY = os.environ.get('SECRET_KEY') or 'gjr39dkejn344_!67#'
    SQLALCHEMY_DATABASE_URI = os.environ.get('DATABASE_URL') or \
        'sqlite:///' + os.path.join(basedir, './db/app.db')
    SQLALCHEMY_TRACK_MODIFICATIONS = False
```

Here is my app/models.py:

```
from app import db
class Request(db.Model):
    id           = db.Column(db.Integer, primary_key=True)
    request_text = db.Column(db.String(400))  
    added_date   = db.Column(db.String(400))  
    added_from   = db.Column(db.String(400))  
    shazaam_id   = db.Column(db.String(400), index=True)           
    yt_title     = db.Column(db.String(400))  
    yt_link      = db.Column(db.String(400))  
    yt_id        = db.Column(db.String(400), index=True)  
    music_artist = db.Column(db.String(400))      
    music_title  = db.Column(db.String(400))      
    music_genre  = db.Column(db.String(400))      
    status       = db.Column(db.String(400))           


    def __repr__(self):
        return self._repr(id=self.id,
                          request_text=self.request_text,
                          added_date=self.added_date)
```

and here is my main script inside the root folder:
```
#!/bin/env python
from app import create_app, socketio
app = create_app(debug=True)
if __name__ == '__main__':
    socketio.run(app)
```

**Comment by miguelgrinberg** [comment, marked as answer]
The `flask run` command runs the application. This command has nothing to do with the database, so what you are seeing is the expected behavior.

**Reply by ManiMatter** [reply]
hi Miguel, 

many thanks for your reply. Maybe then I am a bit puzzled what these two commands should do:
```
 db.init_app(app)
 migrate.init_app(app, db)
```

I would have thought that when I run the application (with flask run), these two commands would 
- the first time I run the app: generate my db (since the db does not yet exist)
- and later: upgrade the db in case there are changes to it

Many thanks again for sharing your insights.

**Reply by miguelgrinberg** [reply]
The `db.init_app()` function registers the Flask-SQLAlchemy extension with your application. The `migrate.init_app()` registers the Flask-Migrate extension with your application.

The database creation, migration and updates are handled through the `flask db ...` set of commands.

**Reply by ManiMatter** [reply]
Thank you for the explanation and your time. Really appreciated.
